docs/ref/models/fields.txt

```diff
@@ -1031,6 +1031,59 @@ are converted to lowercase.
 If you allow for blank values, you have to allow for null values since blank
 values are stored as null.
 
+``JSONField``
+-------------
+
+.. class:: JSONField(encoder=None, decoder=None, **options)
+
+A field for storing JSON encoded data. In Python the data is represented in
+its Python native format: dictionaries, lists, strings, numbers, booleans
+and ``None``.
+
+.. attribute:: JSONField.encoder
+
+    An optional JSON-encoding class to serialize data types not supported
+    by the standard JSON serializer (``datetime``, ``uuid``, etc.). For
+    example, you can use the
+    :class:`~django.core.serializers.json.DjangoJSONEncoder` class.
+
+    If ``None`` is specified, it defaults to :py:class:`json.JSONEncoder`.
+
+.. attribute:: JSONField.decoder
+
+    An optional JSON-decoding class to deserialize the value retrieved from
+    the database. The value will be in the format chosen by the custom encoder
+    (most often a string). Your deserialization may need to account for the
+    fact that you can't be certain of the input type. For example, you run the
+    risk of returning a ``datetime`` that was actually a string that just
+    happened to be in the same format chosen for ``datetime``\s.
+
+    If ``None`` is specified, it defaults to :py:class:`json.JSONDecoder`.
+
+If you give the field a :attr:`~django.db.models.Field.default`, ensure
+it's an immutable or callable object such as ``dict`` (for an empty default)
+or a callable that returns a dict (such as a function). Incorrectly using
+``default={}`` creates a mutable default that is shared between all instances
+of ``JSONField``.
```

Comment:
This reads like `dict` is the only storage type, but it's not. Suggested rewording:

```
If you give the field a :attr:`~django.db.models.Field.default`, ensure
it's an immutable object, such as a ``str``, or a callable object that returns
a fresh mutable object each time, such as ``dict`` or a function. Providing a
mutable default object like ``default={}`` or ``defaualt=[]`` shares the one
object between all model instances. There is a system check to guard against
this.
```

Reply:
Good call. I've copied your rewording (and fixed `defaualt` typo). The system check only produces a warning and not an error, though. Is "There is a system check to guard against this." sufficient?

Reply:
Perhaps "for this" instead of "to guard against this" then?

Reply:
Sure, that'll do.